sdcm/db_log_reader.py

```diff
@@ -125,6 +127,9 @@ def _read_and_publish_events(self) -> None:
                     if json_log:
                         continue
 
+                    if match := self.BUILD_ID_REGEX.search(line):
```

Comment:
we could do some perf optimizations as this will be executed for each line:
1. limit line length (not to parse long lines)
2. Optimize regex to be faster, e.g. `re.compile(r'build-id ([a-f0-9]{40}) starting')` (is ignorecase required?)

Reply:
> we could do some perf optimizations as this will be executed for each line:
> 
> 1. limit line length (not to parse long lines)

we already throw out 5K lines

> 2. Optimize regex to be faster, e.g. `re.compile(r'build-id ([a-f0-9]{40}) starting')` (is ignorecase required?)

ignore case isn't required


```python
import re
import timeit

BUILD_ID_REGEX = re.compile(r'build-id\s(.*?)\sstarting\s\.\.\.', flags=re.IGNORECASE)

BUILD_ID_REGEX_1 = re.compile(r'build-id\s(.*?)\sstarting\s\.\.\.')

BUILD_ID_REGEX_2 = re.compile(r'build-id\s([a-f0-9]{40})\sstarting')

l = "abcdefghij1234567890" * 256
res = timeit.timeit('BUILD_ID_REGEX.search(l)', 'from __main__ import BUILD_ID_REGEX, l')
print(res)


res = timeit.timeit('BUILD_ID_REGEX_1.search(l)', 'from __main__ import BUILD_ID_REGEX_1, l')
print(res)

res = timeit.timeit('BUILD_ID_REGEX_2.search(l)', 'from __main__ import BUILD_ID_REGEX_2, l')
print(res)
```


```
🟢 ❯ python test_123.py
26.814458078995813
1.6338670099939918
1.613332103996072
```

Reply:
remove the `re.IGNORECASE`

the rest is a bit less meaningful